src/api/module/stat/repository/indexeddb/index.ts

```diff
@@ -1,7 +1,48 @@
+import * as entityHistory from '@/api/module/history/entity';
 import * as entity from '@/api/module/stat/entity';
+import * as entityJuz from '@/api/shared/entity/juz';
+import * as entitySurah from '@/api/shared/entity/surah';
+
+import * as calculate from './calculate';
 
 export * from '@/api/module/stat/repository/indexeddb/calculate';
 
+export const getHistoryStat = (history: entityHistory.History): entity.HistoryStat => {
+  const lines: number = calculate.calculateTotalLinesFromHistory(history);
+  return {
+    juz: entityJuz.getTotalJuzFromLines(lines),
+    ayah: getTotalAyah(history),
+    lines: lines,
+  };
+};
+
+const getTotalAyah = (history: entityHistory.History): number => {
+  let totalAyah: number = 0;
+
+  switch (history.historyType) {
+    case entityHistory.HistoryType.Juz:
+      // @ts-expect-error known type
+      const juz: entityJuz.JuzType = entityJuz.getJuzById(history.juz);
+      // eslint-disable-next-line @typescript-eslint/typedef
+      for (let i = juz.startSurah; i <= juz.endSurah; i++) {
+        // @ts-expect-error known type
+        totalAyah += entitySurah.getSurahById(i)?.totalAyah;
```

Comment:
```suggestion
        if (entitySurah.getSurahById(i)) {
           totalAyah += entitySurah.getSurahById(i).totalAyah;
        }
```

Reply:
any concern for this? because I think both will yield same result

even safer variation may look like this, so it only call getSurahById() once

```js
const surah = entitySurah.getSurahById(i);
if (surah && typeof surah.totalAyah === 'number') {
   totalAyah += surah.totalAyah;
}
```

but this code should be fine I think and will add totalAyah with 0 if getSurahById() is undefined cmiiw

```js
        totalAyah += entitySurah.getSurahById(i)?.totalAyah;
```